conjure-java-core/src/main/java/com/palantir/conjure/java/util/ErrorGenerationUtils.java

```diff
@@ -0,0 +1,255 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.util;
+
+import com.google.common.base.CaseFormat;
+import com.google.common.collect.ImmutableList;
+import com.palantir.conjure.java.ConjureAnnotations;
+import com.palantir.conjure.java.Options;
+import com.palantir.conjure.java.types.Expressions;
+import com.palantir.conjure.java.types.SafetyEvaluator;
+import com.palantir.conjure.java.types.TypeMapper;
+import com.palantir.conjure.spec.ConjureDefinition;
+import com.palantir.conjure.spec.EndpointError;
+import com.palantir.conjure.spec.ErrorDefinition;
+import com.palantir.conjure.spec.ErrorNamespace;
+import com.palantir.conjure.spec.ErrorTypeName;
+import com.palantir.conjure.spec.FieldDefinition;
+import com.palantir.conjure.spec.LogSafety;
+import com.palantir.javapoet.AnnotationSpec;
+import com.palantir.javapoet.ClassName;
+import com.palantir.javapoet.JavaFile;
+import com.palantir.javapoet.MethodSpec;
+import com.palantir.javapoet.ParameterSpec;
+import com.palantir.javapoet.TypeName;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import javax.lang.model.element.Modifier;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.function.TriFunction;
+
+public final class ErrorGenerationUtils {
+    public static MethodSpec privateConstructor() {
+        return MethodSpec.constructorBuilder().addModifiers(Modifier.PRIVATE).build();
+    }
+
+    public record DeclaredEndpointErrors(Set<ErrorTypeName> errors) {
+        public static DeclaredEndpointErrors from(ConjureDefinition definition) {
+            return new DeclaredEndpointErrors(definition.getServices().stream()
+                    .flatMap(service -> service.getEndpoints().stream())
+                    .flatMap(endpoint -> endpoint.getErrors().stream())
+                    .map(EndpointError::getError)
+                    .collect(Collectors.toSet()));
+        }
+
+        public boolean contains(ErrorDefinition errorDefinition) {
+            com.palantir.conjure.spec.TypeName errorName = errorDefinition.getErrorName();
+            ErrorTypeName errorTypeName = ErrorTypeName.builder()
+                    .name(errorName.getName())
+                    .package_(errorName.getPackage())
+                    .namespace(errorDefinition.getNamespace())
+                    .build();
+            return errors.contains(errorTypeName);
+        }
+    }
+
+    public record ErrorDefinitionsByPackageAndNamespace(
+            Map<String, Map<ErrorNamespace, List<ErrorDefinition>>> packageToNamespacedErrorDefs) {
```

Comment:
I think this is still a bit of a legibility issue -- we've wrapped the nested generics, but haven't eliminated them.

Would it be reasonable to replace this with a `List<NamespacedErrors>` where `NamespacedErrors` is roughly shaped:
```java
record NamespacedErrors(String javaPackage, ErrorNamespace namespace, List<ErrorDefinition> errors) {}
```

Building the model would require some grouping with similar maps, but the data we're left with to hand the generator is a bit easier to understand. Also, when we build this data model, we can start with something like `record ErrorGroup(String javaPackage, ErrorNamespace namespace)` as the key,  in order to avoid multiple nested maps. We don't need to group the same package together, it's only errors with the same package and namespace in common.

Reply:
Yep, that's cleaner. Updated.